File: C++/moveZerosToEnd.cpp
#include <iostream>

void moveZerosToEnd(int arr[], int size) {
    int nonZeroIndex = 0; // Index to keep track of non-zero elements

    // Traverse the array and move non-zero elements to the front
    for (int i = 0; i < size; i++) {
        if (arr[i] != 0) {
            arr[nonZeroIndex] = arr[i];
            nonZeroIndex++;
        }
    }

    // Fill the remaining elements with zeros
    while (nonZeroIndex < size) {
        arr[nonZeroIndex] = 0;
        nonZeroIndex++;
    }
}

int main() {
    int arr[] = {0, 2, 0, 4, 0, 6, 0, 8, 10, 0};
    int size = sizeof(arr) / sizeof(arr[0]);

    moveZerosToEnd(arr, size);

    // Print the modified array
    for (int i = 0; i < size; i++) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: Java/MoveZerosToEnd.java
public class MoveZerosToEnd {
    public static void main(String[] args) {
        int[] arr = {0, 2, 0, 4, 0, 6, 0, 8, 10, 0};

        int n = arr.length;

        // Move non-zero elements to the front
        int index = 0;
        for (int i = 0; i < n; i++) {
            if (arr[i] != 0) {
                arr[index] = arr[i];
                index++;
            }
        }

        // Fill the rest of the array with zeros
        while (index < n) {
            arr[index] = 0;
            index++;
        }

        // Print the modified array
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
